### PR TITLE
Add support for enterprise consul packages and private repos

### PR DESCRIPTION
**Pull request by roberteckert**

### CODE REVIEW

File: libraries/consul_installation_binary.rb [hunk 1]
```diff
@@ -86,11 +87,14 @@ def consul_program
       end
 
       def self.binary_basename(node, resource)
+        new_filename = node['consul']['enterprise'] ? 'consul-enterprise' : 'consul'
```
Comment (chuckyz) [inline libraries/consul_installation_binary.rb]: Is there any reason to use `new_filename` over `filename` ?

File: libraries/consul_installation_binary.rb [hunk 2]
```diff
@@ -86,11 +87,14 @@ def consul_program
...
+        new_version = node['consul']['enterprise'] ? "#{resource.version}%2bent" : resource.version
```
Comment (chuckyz) [inline libraries/consul_installation_binary.rb]: Same here for `new_version` vs. `version`

File: libraries/consul_installation_binary.rb (hunk 1 above)
Comment (roberteckert) [reply libraries/consul_installation_binary.rb]: I can use the shorter nomenclature.

File: libraries/consul_installation_binary.rb (hunk 2 above)
Comment (roberteckert) [reply libraries/consul_installation_binary.rb]: I can use the shorter nomenclature.

File: libraries/consul_installation_binary.rb (hunk 1 above)
Comment (roberteckert) [reply libraries/consul_installation_binary.rb]: It has been updated.

File: libraries/consul_installation_binary.rb (hunk 2 above)
Comment (roberteckert) [reply libraries/consul_installation_binary.rb]: It has been updated.

File: libraries/consul_installation_binary.rb (hunk 1 above)
Comment (chuckyz) [reply libraries/consul_installation_binary.rb]: Was just trying to understand the importance of `new_`

**Comment by roberteckert** [comment]
Forgot to increment version, so I pushed to update the PR.

**Comment by roberteckert** [comment]
Some tabs snuck in, and messed up alignment Replaced w/ whitespace.

**Comment by roberteckert** [comment]
Nomad is `+ent`, while consul is `+prem`.

**Comment by roberteckert** [comment]
*sigh* actually use the filename that is declared.

**Comment by roberteckert** [comment]
OK, now kitchen tests pass, so I'm more confident in this. I'm going to merge it since previous versions were approved, and the changes between the approved versions and this are fairly small.

**Pull request merged**